File: Prototype3/Frontend/tapatapp_flutter/lib/core/models/tap.dart
import 'package:json_annotation/json_annotation.dart';

part 'tap.g.dart';

@JsonSerializable()
class Tap {
  final int id;
  @JsonKey(name: 'child_id')
  final int childId;
  @JsonKey(name: 'status_id')
  final int statusId;
  @JsonKey(name: 'user_id')
  final int userId;
  final String init; // Hora inicio
  final String end; // Hora fin

  Tap({
    required this.id,
    required this.childId,
    required this.statusId,
    required this.userId,
    required this.init,
    required this.end,
  });

  factory Tap.fromJson(Map<String, dynamic> json) => _$TapFromJson(json);
  Map<String, dynamic> toJson() => _$TapToJson(this);
}

File: ProtypePruebas/Backend/Backend/models/tap.py
from datetime import datetime

class Tap:
    def __init__(self, id, child_id, status_id, user_id, init, end, created_at=None):
        self.id = id
        self.child_id = child_id
        self.status_id = status_id
        self.user_id = user_id
        self.init = init
        self.end = end
        self.created_at = created_at if created_at else datetime.now()

    def to_dict(self):
        return {
            "id": self.id,
            "child_id": self.child_id,
            "status_id": self.status_id,
            "user_id": self.user_id,
            "init": self.init,
            "end": self.end,
            "created_at": self.created_at.isoformat() if self.created_at else None
        }